src/js/common/schemaform/FormPage.jsx

```diff
@@ -54,12 +55,16 @@ class FormPage extends React.Component {
   }
 
   onChange(formData) {
-    this.props.setData(formData);
+    let newData = formData;
+    if (this.props.route.pageConfig.showPagePerItem) {
+      newData = _.set([this.props.route.pageConfig.arrayPath, this.props.params.index], formData, this.props.form.data);
```

Comment:
Perhaps this logic could be pulled out into a helper and documented, it appears you're setting state by creating an array with 'children' and the param.index, creating a key based on the changed form data, and assigning the value to be the relevant config for the page, as described in `config/form.js`?